### PR TITLE
Fix logger messages

### PR DESCRIPTION
**Pull request by svilenvul**
Found by https://github.com/openhab/static-code-analysis

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java [hunk 1]
```diff
@@ -263,7 +263,7 @@ private void informHandlerAboutLinkedChannel(Thing thing, Channel channel) {
             try {
                 handler.channelLinked(channel.getUID());
             } catch (Exception ex) {
-                logger.error("Exception occurred while informing handler:" + ex.getMessage(), ex);
+                logger.error("Exception occurred while informing handler:{}", ex.getMessage(), ex);
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java]: You might want to add the missing space after the colon here as well.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java [hunk 2]
```diff
@@ -282,7 +282,7 @@ private void informHandlerAboutUnlinkedChannel(Thing thing, Channel channel) {
             try {
                 handler.channelUnlinked(channel.getUID());
             } catch (Exception ex) {
-                logger.error("Exception occurred while informing handler:" + ex.getMessage(), ex);
+                logger.error("Exception occurred while informing handler:{}", ex.getMessage(), ex);
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java]: You might want to add the missing space after the colon here as well.

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/DigitalSTROMHandlerFactory.java [hunk 3]
```diff
@@ -59,13 +59,13 @@ public Thing createThing(ThingTypeUID thingTypeUID, Configuration configuration,
         if (BridgeHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
             ThingUID dSSUID = getBridgeThingUID(thingTypeUID, thingUID, configuration);
             if (dSSUID != null) {
-                logger.info(dSSUID.toString());
+                logger.info("{}", dSSUID.toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/DigitalSTROMHandlerFactory.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/SceneDiscoveryService.java [hunk 4]
```diff
@@ -62,8 +62,8 @@ public SceneDiscoveryService(BridgeHandler bridgeHandler, ThingTypeUID supported
      */
     @Override
     public void deactivate() {
-        logger.debug("deactivate discovery service for scene type " + sceneType + " remove thing tyspes "
-                + super.getSupportedThingTypes().toString());
+        logger.debug("deactivate discovery service for scene type {} remove thing tyspes {}", sceneType,
+                super.getSupportedThingTypes().toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/SceneDiscoveryService.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/manager/impl/DeviceStatusManagerImpl.java [hunk 5]
```diff
@@ -198,8 +198,8 @@ public void run() {
                         if (trashDevices.isEmpty()) {
                             currentDevice.setConfig(config);
                             strucMan.addDeviceToStructure(currentDevice);
-                            logger.debug("trashDevices are empty, add Device with dSID "
-                                    + currentDevice.getDSID().toString() + " to the deviceMap!");
+                            logger.debug("trashDevices are empty, add Device with dSID {} to the deviceMap!",
+                                    currentDevice.getDSID().toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/manager/impl/DeviceStatusManagerImpl.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/manager/impl/DeviceStatusManagerImpl.java [hunk 6]
```diff
@@ -246,7 +246,7 @@ public void run() {
 
                 if (!sceneMan.scenesGenerated()
                         && !sceneMan.getManagerState().equals(ManagerStates.GENERATING_SCENES)) {
-                    logger.debug(sceneMan.getManagerState().toString());
+                    logger.debug("{}", sceneMan.getManagerState().toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/manager/impl/DeviceStatusManagerImpl.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/sensorJobExecutor/AbstractSensorJobExecutor.java [hunk 7]
```diff
@@ -67,7 +67,7 @@ public void run() {
                 sensorJob.execute(dSAPI, connectionManager.getSessionToken());
             }
             if (circuit.noMoreJobs()) {
-                logger.debug("no more jobs... stop circuit schedduler with id = " + circuit.getMeterDSID().toString());
+                logger.debug("no more jobs... stop circuit schedduler with id = {}", circuit.getMeterDSID().toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/sensorJobExecutor/AbstractSensorJobExecutor.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/sensorJobExecutor/CircuitScheduler.java [hunk 8]
```diff
@@ -88,12 +88,12 @@ public void addSensorJob(SensorJob sensorJob) {
...
-                logger.debug("Add sensorJob: " + sensorJob.toString() + "to circuitScheduler: " + this.getMeterDSID());
+                logger.debug("Add sensorJob: {} to circuitScheduler: {}", sensorJob.toString(), this.getMeterDSID());
             } else if (checkSensorJobPrio(sensorJob)) {
-                logger.debug("add sensorJob: " + sensorJob.toString() + "with higher priority to circuitScheduler: "
-                        + this.getMeterDSID());
+                logger.debug("add sensorJob: {} with higher priority to circuitScheduler: {}", sensorJob.toString(),
+                        this.getMeterDSID());
             } else {
-                logger.debug("sensorJob: " + sensorJob.getDSID().toString() + " allready exist with a higher priority");
+                logger.debug("sensorJob: {} allready exist with a higher priority", sensorJob.getDSID().toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/sensorJobExecutor/CircuitScheduler.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/sensorJobExecutor/sensorJob/impl/SceneConfigReadingJob.java [hunk 9]
```diff
@@ -50,8 +50,8 @@ public void execute(DsAPI digitalSTROM, String token) {
 
         if (sceneConfig != null) {
             device.addSceneConfig(sceneID, sceneConfig);
-            logger.debug("UPDATED scene configuration for dSID: " + this.device.getDSID() + ", sceneID: " + sceneID
-                    + ", configuration: " + sceneConfig.toString());
+            logger.debug("UPDATED scene configuration for dSID: {}, sceneID: {}, configuration: {}",
+                    this.device.getDSID(), sceneID, sceneConfig.toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/sensorJobExecutor/sensorJob/impl/SceneConfigReadingJob.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/serverConnection/impl/JSONResponseHandler.java [hunk 10]
```diff
@@ -40,8 +40,8 @@ public static boolean checkResponse(JsonObject jsonResponse) {
             return jsonResponse.get(JSONApiResponseKeysEnum.RESPONSE_OK.getKey()).toString()
                     .equals(JSONApiResponseKeysEnum.RESPONSE_SUCCESSFUL.getKey());
         } else {
-            logger.error("JSONResponseHandler: error in json request. Error message : "
-                    + jsonResponse.get(JSONApiResponseKeysEnum.RESPONSE_MESSAGE.getKey()).toString());
+            logger.error("JSONResponseHandler: error in json request. Error message : {}",
+                    jsonResponse.get(JSONApiResponseKeysEnum.RESPONSE_MESSAGE.getKey()).toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/serverConnection/impl/JSONResponseHandler.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/structure/scene/sceneEvent/EventListener.java [hunk 11]
```diff
@@ -169,7 +169,7 @@ private void handleEvent(JsonArray array) {
         if (array.size() > 0) {
             Event event = new JSONEventImpl(array);
             for (EventItem item : event.getEventItems()) {
-                logger.info(item.getProperties().toString());
+                logger.info("{}", item.getProperties().toString());
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/structure/scene/sceneEvent/EventListener.java]: remove the `toString()`

File: extensions/binding/org.eclipse.smarthome.binding.ntp/src/main/java/org/eclipse/smarthome/binding/ntp/handler/NtpHandler.java [hunk 12]
```diff
@@ -152,7 +152,7 @@ public void initialize() {
             startAutomaticRefresh();
 
         } catch (Exception ex) {
-            String msg = "Error occurred while initializing NTP handler: " + ex.getMessage();
+            String msg = "Error occurred while initializing NTP handler: {}" + ex.getMessage();
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.ntp/src/main/java/org/eclipse/smarthome/binding/ntp/handler/NtpHandler.java]: Err, that won't work here as this is just the String variable, not the logging line.

File: extensions/io/org.eclipse.smarthome.io.javasound/src/main/java/org/eclipse/smarthome/io/javasound/internal/AudioPlayer.java [hunk 13]
```diff
@@ -70,7 +70,7 @@ public void run() {
                 mixer = AudioSystem.getMixer(mixerInfo[cnt]);
                 Line.Info[] lineInfos = mixer.getSourceLineInfo();
                 for (Info lineInfo : lineInfos) {
-                    logger.info(lineInfo.toString());
+                    logger.info("{}", lineInfo.toString());
```
Comment (kaikreuzer) [inline extensions/io/org.eclipse.smarthome.io.javasound/src/main/java/org/eclipse/smarthome/io/javasound/internal/AudioPlayer.java]: remove the `toString()`

**Comment by svilenvul** [comment]
Thanks, the requested changes are applied.

**Pull request merged**